persistence/repository/CRUDRepository.go

```diff
@@ -0,0 +1,22 @@
+package repository
+
+import (
+	"github.com/Masterminds/squirrel"
+	"github.com/ditrit/badaas/services/httperrors"
+)
+
+var (
+	// HTTP Error record already exists
+	ErrAlreadyExists = httperrors.NewErrorNotFound("model instance", "the model instance is not in the database")
+)
+
+// Generic CRUD Repository
+type CRUDRepository[T Tabler] interface {
+	Create(instance *T) httperrors.HTTPError
+	Delete(instance *T) httperrors.HTTPError
+	Save(instance *T) httperrors.HTTPError
+	GetByID(id uint) (*T, httperrors.HTTPError)
+	GetAll() ([]*T, httperrors.HTTPError)
```

Comment:
we want a sort option and a filter too, the purpose of this method is to not use the `limit` and `offset`